server/errors/api-error.js

```diff
@@ -0,0 +1,23 @@
+const AbstractError = require('../../abstract-error')
+const httpStatus = require('http-status')
+
+/**
+ * Class representing an API error.
+ * @extends AbstractError
+ */
+class ApiError extends AbstractError {
+  /**
+   * Creates an API error.
+   * @param {string} message - Error message.
+   * @param {number} status - HTTP status code.
+   * @param {boolean} isPublic - Whether the error message should be public.
+   */
+  constructor (message, status = httpStatus.INTERNAL_SERVER_ERROR,
```

Comment:
It is not the opposite? If the arguments with default values are in the end, we can instantiate `ApiError` as:
`new ApiError('message')`